### PR TITLE
M2 feature

### PR DESCRIPTION
**Pull request by SuyashFowdar**
In this feature the following have been done.

- Create App, ButtonPanel and Button components
- Add buttons and symbol

Thank you.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import ButtonPanel from './ButtonPanel';
+
+const App = () => (
+  <div>
+    <div>Display</div>
+    <ButtonPanel />
+  </div>
```
Comment (SpaYco) [inline src/App.js]: * [x] it should be placed in `src/components` directory ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/App.png))
* [x] it should render the `Display` component ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/appFunctionalComponent.png))
* [x] it should have children components wrapped in [React.Fragment](https://reactjs.org/docs/fragments.html#short-syntax) short syntax ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/ReactFragment.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896092/))

File: src/ButtonPanel.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0];
+const symbols = ['+', '-', '*', '/', '='];
+
+const ButtonPanel = () => (
+  <div>
+    <div>Button Panel</div>
+    <Button numbers={numbers} symbols={symbols} />
+  </div>
+);
```
Comment (SpaYco) [inline src/ButtonPanel.js]: - [x] it includes all the following buttons divided in the groups below: ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/buttonPanel/buttons.png))

* Group 1: AC, +/-, %, ÷
* Group 2: 7, 8, 9, X
* Group 3: 4, 5, 6, -
* Group 4: 1, 2, 3, +
* Group 5: 0, ., =


the `Button` component should be used **once** per each symbol

File: src/Button.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  const numberList = numbers.map(number => <button type="button" key={number.toString()}>{number}</button>);
+  const symbolList = symbols.map(symbol => <button type="button" key={symbol.toString()}>{symbol}</button>);
+
+  return (
+    <div>
+      <div>{numberList}</div>
+      <div>{symbolList}</div>
+    </div>
+  );
+};
```
Comment (SpaYco) [inline src/Button.js]: - [x] don't map the symbols, instead, you should get **one** symbol as a prop that gets rendered as a `<button>` tag

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

##### `Display` component

* [x] it is placed in `src/components` directory ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/app/addToComponent.png))
* [x] it accepts `the result of the calculation` as an incoming prop ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/incomingProps.png))
* [x] its result prop should be a String with default value = 0 ([should be String screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/validateAsString.png)) ([default props screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/calculator/milestone2/display/DefaultProp.png))

- [x] more below 👇 

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

File: src/components/App.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ButtonPanel from './ButtonPanel';
+import Display from './Display';
+
+const App = () => (
+  <div>
+    <Display result="0" />
+    <ButtonPanel />
+  </div>
```
Comment (Torres-ssf) [inline src/components/App.js]: - [x] App component should have its children components wrapped in React.Fragment short syntax:
```JS
const App = () => (
  <>
    <Display result="0" />
    <ButtonPanel />
  </>
```
Screenshot from 2021-03-10 12-11-58

- [x] There is no need to pass the `result` prop to the Display component yet. That's why we can declare a default value for it inside the component.

File: src/components/Display.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  return (
+    <div>
+      <div>{result}</div>
+    </div>
+  );
+};
+
+Display.propTypes = {
+  result: PropTypes.string.isRequired,
+};
```
Comment (Torres-ssf) [inline src/components/Display.js]: - [x] The `result` prop should be a String with default value = '0'
```JS
Display.defaultProps = {
  result: '0',
};

Button.propTypes = {
  result: PropTypes.string,
};
```
Screenshot from 2021-03-10 12-17-08
Screenshot from 2021-03-10 12-17-38

**Review by Torres-ssf — CHANGES_REQUESTED** [review]
### Changes required :muscle: 
Hi :hand: @krishnzzz 
Awesome job on this milestone so far. but we still need to make some changes before moving on.

- [x] Check my comments below.

**OPTIONAL**
- [x] Awesome job add descriptive information about the project to your README file. This is really good because increases the professionalism of your repository. However, the text generated by create-react-app it's still there and we can safely remove it. 

Happy coding :computer: 
**Sérgio Torres**

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: 
Hi again @krishnzzz :hand: 
Great job implementing the requested changes.
You may merge the PR.
Good luck with the next milestone.

Happy coding :computer: 
**Sérgio Torres**

**Pull request merged**